.github/workflows/build-release-binaries.yml

```diff
@@ -0,0 +1,101 @@
+      run: |
+        VERSION=$(grep 'var version' app/version/version.go | cut -d'"' -f2)
+        echo "APP_VERSION=$VERSION" >> $GITHUB_ENV
+        echo "RELEASE_VERSION=$VERSION" >> $GITHUB_ENV
+
+    - name: Set version from git tag
+      if: github.ref_type == 'tag'
+      run: |
+        echo "APP_VERSION=${{ github.ref_name }}" >> $GITHUB_ENV
+        echo "RELEASE_VERSION=${{ github.ref_name }}" >> $GITHUB_ENV
+        echo "LDFLAGS_VALUE=-X github.com/obolnetwork/charon/app/version.version=${{ github.ref_name }}" >> $GITHUB_ENV
+
+    - name: Set build flags for branch
+      if: github.ref_type == 'branch'
+      run: |
+        echo "LDFLAGS_VALUE=-X github.com/obolnetwork/charon/app/version.version=${{ env.APP_VERSION }}" >> $GITHUB_ENV
+
+    - name: Build x86_64 binary
+      run: |
+        mkdir -p dist
+        docker run --rm --platform linux/amd64 \
+          -v $(pwd):/workspace -w /workspace \
+          golang:${{ steps.setup-go.outputs.go-version }}-bookworm \
+          bash -c "apt-get update && apt-get install -y build-essential && \
+          env CGO_ENABLED=1 GOOS=linux GOARCH=amd64 \
+          go build -buildvcs=false -tags='netgo osusergo' -ldflags \"${{ env.LDFLAGS_VALUE }} -extldflags '-static'\" \
+          -o dist/charon-${{ env.RELEASE_VERSION }}-linux-amd64"
+
+    - name: Build ARM64 binary (Emulated)
+      run: |
+        # Run build inside an emulated ARM64 container
+        docker run --rm --platform linux/arm64 \
+          -v $(pwd):/workspace -w /workspace \
+          golang:${{ steps.setup-go.outputs.go-version }}-bookworm \
+          bash -c "apt-get update && \
+                   # Install native build tools for ARM64 container
+                   apt-get install -y build-essential && \
+          # Set target GOARCH, but don't need specific CC (native compiler is used)
+          env CGO_ENABLED=1 GOOS=linux GOARCH=arm64 \
+          go build -buildvcs=false -tags='netgo osusergo' -ldflags \"${{ env.LDFLAGS_VALUE }} -extldflags '-static'\" \
+          -o dist/charon-${{ env.RELEASE_VERSION }}-linux-arm64"
+
+    - name: Create release archives
+      run: |
+        cd dist
+        tar czf charon-${{ env.RELEASE_VERSION }}-linux-amd64.tar.gz charon-${{ env.RELEASE_VERSION }}-linux-amd64
+        tar czf charon-${{ env.RELEASE_VERSION }}-linux-arm64.tar.gz charon-${{ env.RELEASE_VERSION }}-linux-arm64
+        echo "# Linux AMD64 (x86_64)" > checksums.txt
+        sha256sum charon-${{ env.RELEASE_VERSION }}-linux-amd64.tar.gz >> checksums.txt
+        echo -e "\n# Linux ARM64 (aarch64)" >> checksums.txt
+        sha256sum charon-${{ env.RELEASE_VERSION }}-linux-arm64.tar.gz >> checksums.txt
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: charon-${{ env.RELEASE_VERSION }}-binaries
+        path: |
+          dist/charon-${{ env.RELEASE_VERSION }}-linux-amd64
+          dist/charon-${{ env.RELEASE_VERSION }}-linux-arm64
```

Comment:
Do we want those uploaded or only the tarballs? I'd say only the tarballs. If we include those as downloads, we should add their checksums as well (but I don't see a reason why we would want them).